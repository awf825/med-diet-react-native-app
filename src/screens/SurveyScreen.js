import axios from 'axios';
import React, { useContext, useEffect, useState } from 'react';
import AppPicker from "../components/AppPicker/AppPicker"
import {
  SafeAreaView,
  FlatList,
  StyleSheet,
  StatusBar,
  Text,
  View
} from 'react-native';
import { BASE_URL } from '../config';

const SurveyScreen = ({ navigation }) => {
    const [questions, setQuestions] = useState([]);

    let options = [];
    for (let i = 0; i <= 30; i++) {
        options.push({
            label: String(i),
            value: i
        })
    }

    useEffect(() => {
        axios.get(`${BASE_URL}/api/questions/`)
        .then(resp => {
            setQuestions(resp.data)
        })
        .catch(err => console.log(err))
    }, [])

    const _renderItem = ({item, index}) => (
        // <FormBlock item={item} index={index} />
        <View style={styles.item}>
            {
                <AppPicker 
                    label={item.question_text}
                    options={options} 
                />
            }
        </View>
    );

    return (
        <SafeAreaView style={styles.container}>
            <FlatList
                testId={"LIST"}
                data={questions}
                renderItem={_renderItem}
                keyExtractor={item => item.question_id}
            />
        </SafeAreaView>
    );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: StatusBar.currentHeight || 0,
  },
  item: {
    // backgroundColor: '#f9c2ff',
    padding: 20,
    marginVertical: 8,
    marginHorizontal: 16,
  },
  title: {
    fontSize: 32,
  },
});

export default SurveyScreen;