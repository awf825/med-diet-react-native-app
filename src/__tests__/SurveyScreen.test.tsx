import { fireEvent, render, screen, userEvent, waitFor } from '@testing-library/react-native';
import React from 'react'
import SurveyScreen from '../screens/SurveyScreen';

import {
    it,
    jest,
    test,
    expect
} from "@jest/globals"

test('Expect the Survey Form to be rendered', async () => {
    const { getByTestId } = render(<SurveyScreen navigation={undefined} />)
    await waitFor(
        () => {
            const submitButton = getByTestId("SUBMIT");
            expect(submitButton).toBeDefined;
        }
    );  
})