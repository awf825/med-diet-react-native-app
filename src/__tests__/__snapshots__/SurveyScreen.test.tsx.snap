// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
      "marginTop": 0,
    }
  }
>
  <RCTScrollView
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`Survey should render tree 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
      "marginTop": 0,
    }
  }
>
  <RCTScrollView
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;
